src/toolkits/object_detection/object_detector.cpp

```diff
@@ -610,9 +610,17 @@ variant_type object_detector::predict(
     flex_list predicted_row_ft;
     flex_list class_labels = read_state<flex_list>("classes");
     for (const image_annotation& each_row : predicted_row) {
-      flex_dict bb_dict = {{"x", each_row.bounding_box.x}, {"y", each_row.bounding_box.y},
-                      {"width", each_row.bounding_box.width},
-                      {"height", each_row.bounding_box.height}};
+      float width_scale = each_row.bounding_box.img_width;
+      float height_scale = each_row.bounding_box.img_height;
+
+      flex_dict bb_dict = {
+          {"x", (each_row.bounding_box.x + each_row.bounding_box.width / 2.) *
```

Comment:
Let’s factor out this kind of logic to `od_data_iterator.hpp`, since that’s where lives the code for converting from the user-provided (`flex_dict`) data

Reply:
the `consumer` is a function of two `std::vector<image_annotation`, so I stored the real image size in `image_annotation` instead.

Reply:
I think it would be clearer to change the `consumer` function to take the image's original dimensions

When in doubt, leave the lower-level code as simple as possible!

Reply:
good!